prob_56.py

```diff
@@ -0,0 +1,78 @@
+
+
+
+#O(1) - space
+#O(n) time
+#passed in leet code
+#logic is that at each level we have the numbe rof elements added and accordingly add the next elements in the list - BFS approach
+# Definition for a binary tree node.
+# class TreeNode(object):
+#     def __init__(self, x):
+#         self.val = x
+#         self.left = None
+#         self.right = None
+from collections import deque
+class Solution:
+    def levelOrder(self, root):
+        res = []
+        if not root:
+            return res
+        q= deque([root])
+        #temp_lst = [] #if we declare here for each level we have to empty this
+        while q:
+            temp_lst = [] #for each level we are creating a new list
+            size = len(q)
+            for i in range(size):
+                curr = q.popleft() #popleft in deques folloew fifo
+                temp_lst.append(curr.val)
+                if curr.left:
+                    q.append(curr.left)
+                if curr.right:
+                    q.append(curr.right)

```

Comment:
is this complete?

Reply:
sorry uploaded different file, changed the file please check